scale/batch/models.py

```diff
@@ -100,6 +100,77 @@ def create_batch_v6(self, title, description, recipe_type, event, definition, co
             BatchMetrics.objects.bulk_create(batch_metrics_models)
 
         return batch
+        
+    def calculate_estimated_recipes(self, batch, definition):
+        """Calculates the estimated number of recipes that will be created for this batch. 
+        This number is calculated by:
+        1. The number of existing recipes for the specific recipe type that are 
+           not currently superseded
+        2. The number of sub-recipes in the recipe
+           These should be filtered if not changed/marked for re-run?
+           
+        """
+        
+        # If this is a previous batch, use the previous batch total
+        if batch.superseded_batch:
+            return batch.superseded_batch.recipes_total
+        
+        # No files defined to run on, so no recipes will be created
+        if not definition.dataset:
+            return 0
+
+        #: The number of recipes are calculated based on the following:
+        #      - If the dataset has a global parameter matching the input of the 
+        #        recipe type, count the number of files in each dataset member
+        #      - If the dataset has a parameter matching the input of the recipe
+        #        type, count the number of files in each member that matches the parameter
+        
+        from recipe.models import RecipeType
+        from data.models import DataSet, DataSetMember, DataSetFile
+        dataset = DataSet.objects.get(pk=definition.dataset)
+        dataset_definition = dataset.get_definition()
+        recipe_type = RecipeType.objects.get(pk=batch.recipe_type_id)
```

Comment:
use recipe type revision here as the definition could have changed since the batch was created